### PR TITLE
Product Listing, Filtering and Sorting

### PR DESCRIPTION
**Pull request by Tanay-J**
This PR adds a new feature: Product Listing Page
It includes:
1. Section to filter and sort products
   - Sorting by Price (low to high / high to low) 
   - Filtering by Categories
   - Filtering by Price using range slider
   - Filtering by binding type of the book 
   - Sorting by Product Rating
 
2. Section to display sorted and filtered products

Apart from this the PR includes minor tweaks in styles as well.

**Demo**

https://user-images.githubusercontent.com/33786522/159982263-af468e4c-c9b3-49f6-89a2-e4559a58717f.mp4

### CODE REVIEW

File: src/utils/getFinalProducts.jsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  };
+  
+  const getSortedProducts = (state, products) => {
+    switch (state.sorting) {
+      case "lth":
+        return products.sort(
+          (a, b) =>
+            a.price -
+            (a.price * a.discount) / 100 -
+            (b.price - (b.price * b.discount) / 100)
```
Comment (Hamdrive) [inline src/utils/getFinalProducts.jsx]: Better to convert to Number type to ensure no side effects

**Review by Hamdrive — COMMENTED** [review]
Good code written, like the use of destructuring props which are passed to components.

Have suggested changes to use `Number` type when doing sorting and other filters, as direct string comparisons can have side effects

File: src/components/products/FilterAndSort.jsx [hunk 2]
```diff
@@ -1,5 +1,189 @@
...
+              })
+            }
+          />
+          <small>Fast Delivery Only</small>
+        </label>
+      </form>
+
+      <form className="flex flex-col my-xs">
+        <h5 className="my-xs">Format</h5>
+        <label for="pb">
```
Comment (junaidshaikh-js) [inline src/components/products/FilterAndSort.jsx]: In JSX we need to write `htmlFor` instead of `for` as it is a reserved keyword

**Review by junaidshaikh-js — APPROVED** [review]
Well-written code! 

Everything is well structured, a good naming convention is followed. Great Work!

Just need to do minor changes as suggested in the comments.

File: src/components/products/FilterAndSort.jsx (hunk 2 above)
Comment (Tanay-J) [reply src/components/products/FilterAndSort.jsx]: Thanks for pointing out. Changed to ```htmlFor```

File: src/utils/getFinalProducts.jsx (hunk 1 above)
Comment (Tanay-J) [reply src/utils/getFinalProducts.jsx]: Made suggested changes to the code. Thank you for reviewing.

**Pull request merged**